### PR TITLE
external/stdk: fix timer issue, and update patch for stdk

### PR DESCRIPTION
**Pull request by SRCN-LiQin**
1. Fix timer issue in bsp wrapper, which cause the endless log when
exceptional case like no response for confirmation, or AP selection.
2. Fix stdk issue of wrong return value type, which cause the return
value can't be checked normally for NO confirmation case on artik board.
3. add log in esp restart api, for debug convenient.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [9c2da213ea6ae03960c63cca4d72842456c94a53] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [9c2da213ea6ae03960c63cca4d72842456c94a53] - Code Rule Check (C++) OK.

File: external/stdk/tizenrt-wrapper/os/iot_os_util_tizenrt.c [hunk 1]
```diff
@@ -281,14 +281,18 @@ static int check_for_timeout(clock_t *const ptime_out, clock_t *const pticks_to_
 
 void iot_os_timer_count_ms(iot_os_timer timer, unsigned int timeout_ms)
 {
-	((tizenrt_timer *)timer)->ticks_to_wait = MSEC2TICK(timeout_ms); /* convert milliseconds to ticks */
+	((tizenrt_timer *)timer)->ticks_to_wait = (timeout_ms == portMAX_DELAY) ? portMAX_DELAY : MSEC2TICK(timeout_ms); /* convert milliseconds to ticks */
```
Comment (zhouxinhe) [inline external/stdk/tizenrt-wrapper/os/iot_os_util_tizenrt.c]: Should we handle `MSEC2TICK(wait_time_ms)` at line 122 in same way?
How about making a local macro `_MSEC2TICK` to handle portMAX_DELAY logic?

Comment (SRCN-LiQin) [reply external/stdk/tizenrt-wrapper/os/iot_os_util_tizenrt.c]: although stdk have no requirement of this in the function at line 122, but for complete logic, it's better to refine this. thanks for suggestion.

**Comment by seinfra** [comment]
Target : [868daa00eee2aa018e354c097fa13e97a81f0278] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [868daa00eee2aa018e354c097fa13e97a81f0278] - Code Rule Check (C++) OK.

**Pull request merged**